base_placement_plugin/src/create_marker.cpp

```diff
@@ -0,0 +1,313 @@
+    for(int k=0;k<joint_soln.size();k++)
+    {
+      robot_state_->setJointPositions(joint_names[k], &(joint_soln[k])); //Setting joint values to arm joints
+    }
+    robot_state_->update(); //Updating robot state
+    const std::vector<std::string>& arm_link_names = robot_jmp->getLinkModelNames();//Obtaining all the arm link names
+    std::vector<std::string> full_link_names = robot_model_->getLinkModelNames(); //Obtaing all the robot model names
+    int position = std::find(full_link_names.begin(), full_link_names.end(), arm_link_names[0]) -full_link_names.begin() ;
+    std::string arm_parent_link = full_link_names[position-1];
+    std::vector<std::string> full_links;
+    for(int k=0;k<arm_link_names.size();k++)
+    {
+      full_links.push_back(arm_link_names[k]);
+    }
+
+    if(has_joint_model_ee)
+    {
+      for(int k=0;k<ee_links.size();k++)
+      {
+        full_links.push_back(ee_links[k]);
+      }
+    }
+
+    visualization_msgs::MarkerArray full_link_markers;
+    robot_state_->getRobotMarkers( full_link_markers, full_links);//Obtain the current markers
+
+    Eigen::Affine3d tf_root_to_first_link = robot_state_->getGlobalLinkTransform(arm_parent_link);//transform of the parent link
+    Eigen::Affine3d tf_first_link_to_root = tf_root_to_first_link.inverse();//inverse of the transform
+
+    Eigen::Affine3d base_tf;
+    tf::poseMsgToEigen(base_pose, base_tf);//base pose in tf
+    visualization_msgs::InteractiveMarker iMarker; //new interactive marker
+    iMarker.header.frame_id = "base_link";
+    iMarker.pose = base_pose;//the pose of the imarker is the pose of the base
+    iMarker.scale = 0.2;
+    std::string name = "robot_model";
+    std::string description = "robot model";
+    iMarker.name = name+boost::lexical_cast<std::string>(i);
+    iMarker.description = description+boost::lexical_cast<std::string>(i);//each imarker has different name and description
+    visualization_msgs::InteractiveMarkerControl robotModelControl;
+    for(std::size_t j=0; j<full_link_markers.markers.size();j++)
+    {
+      full_link_markers.markers[j].header.frame_id = "base_link";
+      full_link_markers.markers[j].type = visualization_msgs::Marker::MESH_RESOURCE;
+      full_link_markers.markers[j].mesh_use_embedded_materials = true;
+      full_link_markers.markers[j].id = i*(j+5);
+      full_link_markers.markers[j].header.stamp = ros::Time::now();
+      full_link_markers.markers[j].ns = "robot_links";
+      full_link_markers.markers[j].lifetime = ros::Duration(40.0);
+      //Just setting random color.Can be ignored
+      if(no_joint_solution)
+      {
+        full_link_markers.markers[j].color.r = 1.0;
+        full_link_markers.markers[j].color.g = 0.0;
+        full_link_markers.markers[j].color.b = 0.0;
+        full_link_markers.markers[j].color.a = 0.2;
+      }
+      else
+      {
+        if(j==1 || j==2 || j==5 || j==6 || j==9)
```

Comment:
Again, what are these numbers?  Will they work for any robot models or only 6 DOF?